builder/proxmox/step_start_vm.go

```diff
@@ -142,6 +142,10 @@ func generateProxmoxDisks(disks []diskConfig) proxmox.QemuDevices {
 		setDeviceParamIfDefined(devs[idx], "storage_type", disks[idx].StoragePoolType)
 		setDeviceParamIfDefined(devs[idx], "cache", disks[idx].CacheMode)
 		setDeviceParamIfDefined(devs[idx], "format", disks[idx].DiskFormat)
+
+		if devs[idx]["type"] == "scsi" || devs[idx]["type"] == "virtio" {
```

Comment:
Is this ever not the case, since you validate it in `config.Prepare`?

Reply:
My initial thinking was to only send it for disks that support it, but I think you're right, we could remove this check and always send it, albeit with disabled state 🚀 

But first I want to double-check that proxmox will not complain about this parameter being present (but disabled) for disks that do not support it. It shouldn't have any impact, but you never know 🤷

Reply:
Hm, maybe I'm misreading the code? My understanding was that it will only possibly be set for objects in `devs[idx]` that support it, since we validate it earlier, but maybe I'm missing something?

Reply:
Yes, that's correct. I was just thinking about the reverse case where it is not set and this if statement is removed.

Meaning that IO thread parameter would always be present (sent) for all disks, but with a value of false, and I was just wondering whether proxmox will like that if that makes sense

Reply:
If it isn't present at all, `setDeviceParamIfDefined` should do nothing, so that should be covered already

Reply:
The `setDeviceParamIfDefined` only guards against empty strings and `strconv.FormatBool` returns a string that is not empty (e.g. "false", "true") hence it would always add the parameter

Reply:
Ahh, true that. Okay, but then we'll wait to see about your experiment :)

Reply:
Looks like proxmox doesn't like that. I tried creating a VM with a sata disk and `io_thread: false` and got this error:

> sata0.iothread: property is not defined in schema and the schema does not allow additional properties

Had a quick snoop around proxmox api docs and indeed only scsi and virtio devices have `iothread` property defined, so I guess that makes sense

Reply:
Alright, then everything is as it should be :) Thanks for the investigation!